dcos/dcos-ui
# DCOS-12637: add metadata to health check endpoint dropdown

Not all of the possible states were covered in the design Lee attached to the ticket, so here's a spreadsheet showing metadata content for all the states I was able to find: [Health check port metadata](https://docs.google.com/spreadsheets/d/1myz9yy9Xx1H4Vybjl7zC9j7DtZosrZCXRyjevhuFGcU/edit?usp=sharing)

## Testing

Single Container
1. In the create service form, go to the Network tab
2. Add a few endpoints - some with names and some without
3. Go to the Health Checks tab and add a health check
4. Select "HTTP" protocol

The "Select Endpoints" select dropdown should have additional metadata in parentheses in addition to what it was showing before this change.
If an endpoint has a Host Port auto-assigned, it should show the environemnt variable text "$PORT{index}"

The results should be consistent no matter which "Network Type" you select from the dropdown

Multi-container
1. In the create service form, go to the Network tab
2. Add a few endpoints to a container - some with names and some without
3. Go to the Health Checks tab and add a health check to that container
4. Select "HTTP" protocol

The "Select Endpoints" select dropdown should have additional metadata in parentheses in addition to what it was showing before this change.
If an endpoint has a Host Port auto-assigned, it should show the environment variable text "$ENDPOINT_{endpoint name}"

The results should be consistent no matter which "Network Type" you select from the dropdown


## Trade-offs



## Dependencies



## Screenshots
Before:
<img width="953" alt="Screen Shot 2019-05-07 at 8 13 47 PM" src="https://user-images.githubusercontent.com/2313998/57342184-ac408800-710b-11e9-9c31-b44539b753ef.png">

After:
<img width="967" alt="Screen Shot 2019-05-07 at 8 12 50 PM" src="https://user-images.githubusercontent.com/2313998/57342188-b19dd280-710b-11e9-876f-6aa533aaad05.png">
